anatomy_of_variables/Anatomy of Variables.md

```diff
@@ -0,0 +1,229 @@
+
+Class variables work exactly like instance methods. These varibales are denoted with a double `@` sign, i.e `@@`. These variables are  **ONLY** used in **Class Methods** within a class. Just in case, you don't know what a class method is in ruby, let's take a quick look.
+
+The keyword `self` makes a class method quite diferent from an instance method. A class method is also defined by the `def` keyword, then the keyword `self` is prpended to the method name. For example, if we were to rewrite the above example as a class method, the method `greet` will be written as `self.greet`. Class methods, however, can be called directly on the class they belong to. The example below shows a class variable been used in a class method.
+
+```ruby
+Class User
+    @@name = Adebayo
+
+    def self.name
+         puts "Hello #{@@name}"
+    end
+end
+
+User.name  #=> "Hello Adebayo"
+```
+
+**Class Instance Variables**
+
+These variables are a combination of the two we have discussed earlier. They are used as global variables but they are assigned as local variables. This means they are assigned in a class method like instance variables but they can be used in any class methods within the same class.
+
+The example below shows how class instance variables are used.
+
+```ruby
+class Person
+    def self.username(name)
+      @name = name
+    end
+
+    def self.greet
+      "Hello #{@name}"
+    end
+end
+
+Person.username("Adebayo") #=> "Adebayo"
+Person.greet #=> "Hello Adebayo"
+
+```
+
+In the snippet above, `name` method was used to get the person's name and it was assigned to `@name` variable. This variable, however, can be used in any other class method within that class.
+
+One important thing to know is that this kind of variables will not be available in classes that inherit from the class in which they are assigned.
+
+```ruby
+class Property < Person
+    def self.owner
+        "I belong to #{@name}"
+    end
+end
+
+
+Property.owner #=> "I belong to "
+@name = nil
+```
+In other to use this vaiable in this new class, we have to redeclare the variable. A correct way to do this will be;
+
+```ruby
+class Property < Person
+    def self.owner
+        "I belong to #{@name}"
+    end
+end
+
+Property.username("Adebayo") #=> "Adebayo"
+Property.owner #=> "I belong to Adebayo"
+```
+
+## What makes these varables different from each other?
+
+The major differences between these variables wrap around the scope at which they are to be used and the syntax for assigning them.
+
+* The diference between an instance variable and a class instance variable is the scope at which they are been used.
+
+    > Instance Variables are scoped to Instance Methods while Class Instance Variable are scoped to Class Methods.
+
+* The diference between a Class Instance Variable and a Class Variable is the syntax of assinging them.
+
+    > Class instance Variables are assigned with a single `@` symbol while Class Variables are assigned with double `@` symbol, i.e `@@`
```

Comment:
I feel it always makes sense to have a closing remark like concusion or final thoughts. What do you think of having such?